core/arch/x86/decode_table.c

```diff
@@ -7171,15 +7171,15 @@ const instr_info_t vex_W_extensions[][2] = {
      * registers are identical.  We don't bother trying to detect that.
      */
     {OP_vpgatherdd,0x66389018,"vpgatherdd",Vx,Hx,MVd,Hx,xx, mrm|vex|reqp,x,END_LIST},
-    {OP_vpgatherdq,0x66389058,"vpgatherdq",Vx,Hx,MVq,Hx,xx, mrm|vex|reqp,x,END_LIST},
+    {OP_vpgatherdq,0x66389058,"vpgatherdq",Vx,Hx,MVd,Hx,xx, mrm|vex|reqp,x,END_LIST},
```

Comment:
First, any changes here should be synchronized with instr_compute_VSIB_index().

Second, the original looks correct to me: the first letter of the opcode name suffix is the index size, while the second is the memory size.  So "vpgatherdq" is a d-sized index and q-sized memory.  The operand size we store for memory accesses is always the memory size, not the address size, and for VSIB we store a single memory element size.  It looks like we have no inside-opnd_t storage of the index size: that's done by dispatch on opcode it seems.  I have a bunch of notes of discussions on how to handle VSIB: we decided to bail on encoding too much inside the opnd_t I guess.